lib/sections/publish.js

```diff
@@ -1,79 +1,69 @@
-var utils = require(__dirname + '/utils.js');
-var uuid  = require('node-uuid');
-var os    = require('os');
-// var i18n  = require('i18n');
-// var time  = require('time');
+var utils  = require(__dirname + '/utils.js');
+var uuid   = require('node-uuid');
+var os     = require('os');
+var time   = require('time');
 
 var publish = function(eventType, args, callback){
   var self = this;
   var queue = self.options.incommigQueue;
-  var klass = self.options.busDriverClassKey;
-
-  payload = [publishMetadata(eventType, args)];
+  var klass = self.options.busClassKey;
+  var payload = [publishMetadata(eventType, args)];
+  payload[0].bus_class_proxy  = '::QueueBus::Driver';
 
   self.queueObject.enqueue(queue, klass, payload, function(err, toRun){
     if(typeof callback === 'function'){ callback(err, toRun); }
   });
-}
+};
 
 var publishAt = function(timestamp, eventType, args, callback){
-  var self = this;
-  var queue = self.options.incommigQueue;
-  var klass = self.options.busPublisherClassKey;
+  var self    = this;
+  var queue   = self.options.incommigQueue;
+  var klass   = self.options.busClassKey;
+  var payload = [publishMetadata(eventType, args)];
+  payload[0].bus_class_proxy  = '::QueueBus::Publisher';
 
-  payload = [publishMetadata(eventType, args)];
-  if(payload["bus_delayed_until"] == null){
-    payload["bus_delayed_until"] = Math.floor(timestamp/1000);
+  if(payload.bus_delayed_until === undefined){
+    payload.bus_delayed_until = Math.floor(timestamp/1000);
   }
-  delete payload['bus_published_at']; // will get re-added upon re-publish
 
-  self.queueObject.enqueueAt(timestamp, queue, klass, payload, function(){
-    if(typeof callback === 'function'){ callback(); }
+  delete payload.bus_published_at; // will get re-added upon re-publish
+
+  self.queueObject.enqueueAt(timestamp, queue, klass, payload, function(err){
+    if(typeof callback === 'function'){ callback(err); }
   });
-}
+};
 
 var publishIn = function(time, eventType, args, callback){
   var self = this;
   var timestamp = new Date().getTime() + time;
   self.publishAt(timestamp, eventType, args, callback);
-}
+};
 
 var publishMetadata = function(eventType, args){
   var payload = {};
-  if(eventType != null){
-    payload['bus_event_type'] = eventType
+  if(eventType){
+    payload.bus_event_type = eventType;
   } else {
-    payload['bus_event_type'] = null;
+    payload.bus_event_type = null;
+  }
+  payload.bus_published_at = utils.timestamp();
+  payload.bus_id           = utils.timestamp() + "-" + uuid.v4();
+  payload.bus_app_hostname = os.hostname();
+  payload.bus_timezone     = new time.Date().getTimezone();
```

Comment:
be interesting to see if this matches up with the ruby strings. if not, we might want to leave it off.